src/sagemaker/image_uri_config/framework_upgrade.py

```diff
@@ -0,0 +1,146 @@
+
+    Returns:
+        dict: Dictionary of region to registry mapping.
+    """
+    with open("framework-region-registry.json", "r") as f:
+        frameworks = json.load(f)
+        return frameworks[framework]
+
+
+def add_dlc_framework_version(
+    content,  # pylint: disable=W0621
+    framework,  # pylint: disable=W0621
+    short_version,  # pylint: disable=W0621
+    full_version,  # pylint: disable=W0621
+    image_type,  # pylint: disable=W0621
+    processors,  # pylint: disable=W0621
+    py_versions,  # pylint: disable=W0621
+    registries,  # pylint: disable=W0621
+):
+    """Update framework image uri json file with new version information.
+
+    Args:
+        content (dict): Existing framework image uri information read from "<framework>.json" file.
+        framework (str): Framework name (e.g. tensorflow, pytorch, mxnet)
+        short_version (str): Abbreviated framework version (e.g. 1.0, 1.5)
+        full_version (str): Complete framework version (e.g. 1.0.0, 1.5.2)
+        image_type (str): Framework image type, it could be "training", "inference" or "eia"
+        processors (list): Supported processors (e.g. ["cpu", "gpu"])
+        py_versions (list): Supported Python versions (e.g. ["py3", "py37"])
+    """
+    for processor in processors:
+        if processor not in content[image_type]["processors"]:
+            content[image_type]["processors"].append(processor)
+    content[image_type]["version_aliases"][short_version] = full_version
+    repo = "{}-{}{}"
+    if image_type == "eia":
+        repo = repo.format(framework, "inference", "-eia")
+    else:
+        repo = repo.format(framework, image_type, "")
+    add_version = {
+        "registries": registries,
+        "repository": repo,
+        "py_versions": py_versions,
+    }
+    content[image_type]["versions"][full_version] = add_version
+
+
+def update_json(
+    framework, short_version, full_version, image_type, processors, py_versions
+):  # pylint: disable=W0621
+    """Read framework image uri information from json file to a dictionary, update it with new
+    framework version information, then write the dictionary back to json file.
+
+    Args:
+        framework (str): Framework name (e.g. tensorflow, pytorch, mxnet)
+        short_version (str): Abbreviated framework version (e.g. 1.0, 1.5)
+        full_version (str): Complete framework version (e.g. 1.0.0, 1.5.2)
+        image_type (str): Framework image type, it could be "training", "inference" or "eia"
+        processors (str): Supported processors (e.g. "cpu,gpu")
+        py_versions (str): Supported Python versions (e.g. "py3,py37")
+     """
+    filename = "{}.json".format(framework)
+    content = _read_json_to_dict(filename)
+    py_versions = py_versions.split(",")
+    processors = processors.split(",")
+    registries = _read_framework_region_regestries(framework)["registries"]
+    add_dlc_framework_version(
```

Comment:
could we basically default to the most recent version for anything  that's not provided to the user?

Reply:
Removed registries related files and take latest region account mappings, py_versions and repository values.